polyply/tests/test_gen_coords_logic.py

```diff
@@ -47,6 +47,51 @@ def test_no_positions_generated(tmp_path, monkeypatch):
         assert np.all(molecule_out.nodes[node]['position'] ==
                       molecule_in.nodes[node]['position'])
 
+@pytest.mark.parametrize('box_input, box_ref, density, warning', [
+                        # box from input coordinates
+                        (None, np.array([11.0, 11.0, 11.0]), None, None),
+                        # box from input coordinates overwrites
+                        (np.array([5.0, 5.0, 5.0]), np.array([11.0, 11.0, 11.0]), None, "warn1"),
+                        # box from input coordinates and density from CLI
+                        (None, np.array([11.0, 11.0, 11.0]), 1000, "warn2"),
+                        ])
+def test_box_input(tmp_path, monkeypatch, caplog, box_input, box_ref, density, warning):
+    """
+    Here we test that the correct box is chosen, in case there
+    are conflicting inputs.
+    """
+    warnings = {"warn1": ("A box is provided via the -box command line "
+                          "and the starting coordinates. We consider the "
+                          "the box of starting coordinates as correct. "),
+                "warn2": ("A density is provided via the command line, "
+                          "but the starting coordinates define a box."
+                          "Will try to pack all molecules in the box "
+                          "provied with starting coordinates."),}
+
+    monkeypatch.chdir(tmp_path)
```

Comment:
changing to a new temppath is a bit safer since this test generates output files that could interfere with other tests when piled in the same directory. It's kinda same infrastructure as the integration tests.

Reply:
On my machine there is a `/tmp/pytest-of-peterkroon/pytest-1` which contains a whole bunch of folders. So I really don't think this adds anything.

Reply:
Let's see